seqr/utils/elasticsearch/es_search.py

```diff
@@ -213,16 +211,87 @@ def sort(self, sort):
 
         self._search = self._search.sort(*self._sort)
 
-    def filter(self, new_filter):
+    def _filter(self, new_filter):
         self._search = self._search.filter(new_filter)
         return self
 
-    def filter_by_in_silico(self, in_silico_filters):
-        in_silico_filters = {k: v for k, v in in_silico_filters.items() if v is not None and len(v) != 0}
+    def filter_variants(self, inheritance=None, genes=None, intervals=None, rs_ids=None, variant_ids=None, locus=None,
+                        frequencies=None, pathogenicity=None, in_silico=None, annotations=None, annotations_secondary=None,
+                        quality_filter=None, custom_query=None, skip_genotype_filter=False):
+        has_location_filter = genes or intervals or rs_ids or variant_ids
+
+        self._filter_custom(custom_query)
+
+        if has_location_filter:
+            self.filter_by_location(
+                genes=genes, intervals=intervals, rs_ids=rs_ids, variant_ids=variant_ids, locus=locus)
+
+        self._filter_by_frequency(frequencies, pathogenicity=pathogenicity)
+
+        self._filter_by_in_silico(in_silico)
+
+        if quality_filter and quality_filter.get('vcf_filter') is not None:
+            self._filter(~Q('exists', field='filters'))
+
+        annotations = {k: v for k, v in (annotations or {}).items() if v}
+        new_svs = bool(annotations.pop(NEW_SV_FIELD, False))
+        splice_ai = annotations.pop(SPLICE_AI_FIELD, None)
+        self._allowed_consequences = sorted({ann for anns in annotations.values() for ann in anns})
+
+        inheritance_mode = (inheritance or {}).get('mode')
+        inheritance_filter = (inheritance or {}).get('filter') or {}
```

Comment:
Would it better to use `.get('filter', {})`?

Reply:
no, if filter is included and set to null (which can happen from the ui) then everything would break